vcl_templates/www.vcl.erb

```diff
@@ -296,6 +304,11 @@ sub vcl_deliver {
     add resp.http.Set-Cookie = "ABTest-Example=" req.http.GOVUK-ABTest-Example "; expires=" now + 1d;
   }
 
+  # Set the Javascript detection cookie
+  if (req.http.User-Agent !~ "^GOV\.UK Crawler Worker" && req.http.Cookie !~ "JS-Detection") {
+    add resp.http.Set-Cookie = "JS-Detection=" req.http.JS-Detection "; expires=" now + 5w;
```

Comment:
You'll probably need to add `"; path=/"` to the end of the cookie. Otherwise the browser defaults to the path of the individual page, and you'll end up with users with multiple IDs for different parts of the site.

We don't add this to the `ABTest-Example` cookie because it's only used on one page, but I've added it to `ABTest-EducationNavigation` in alphagov/govuk-cdn-config#17.